src/cryptography/x509.py

```diff
@@ -141,6 +142,57 @@ def __len__(self):
         return len(self._attributes)
 
 
+OID_BASIC_CONSTRAINTS = ObjectIdentifier("2.5.29.19")
+
+
+class Extension(object):
+    def __init__(self, oid, critical, value):
+        if not isinstance(oid, ObjectIdentifier):
+            raise TypeError(
+                "oid argument must be an ObjectIdentifier instance."
+            )
+
+        if not isinstance(critical, bool):
+            raise TypeError("critical must be a boolean value")
+
+        self._oid = oid
+        self._critical = critical
+        self._value = value
+
+    oid = utils.read_only_property("_oid")
+    critical = utils.read_only_property("_critical")
+    value = utils.read_only_property("_value")
```

Comment:
Ok, so obviously not that ;-)

On Mon, Mar 23, 2015 at 7:56 PM, Paul Kehrer <notifications@github.com>
wrote:

> In src/cryptography/x509.py
> <https://github.com/pyca/cryptography/pull/1768#discussion_r26993014>:
>
> > +    def __init__(self, oid, critical, value):
> > +        if not isinstance(oid, ObjectIdentifier):
> > +            raise TypeError(
> > +                "oid argument must be an ObjectIdentifier instance."
> > +            )
> > +
> > +        if not isinstance(critical, bool):
> > +            raise TypeError("critical must be a boolean value")
> > +
> > +        self._oid = oid
> > +        self._critical = critical
> > +        self._value = value
> > +
> > +    oid = utils.read_only_property("_oid")
> > +    critical = utils.read_only_property("_critical")
> > +    value = utils.read_only_property("_value")
>
> Yeah, the spec calls it "extnValue" [image: :cry:]
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pyca/cryptography/pull/1768/files#r26993014>.
>



-- 
"I disapprove of what you say, but I will defend to the death your right to
say it." -- Evelyn Beatrice Hall (summarizing Voltaire)
"The people's good is the highest law." -- Cicero
GPG Key fingerprint: 125F 5C67 DFE9 4084